lib/mathjax-helper.coffee

```diff
@@ -43,15 +44,72 @@ module.exports =
     return
 
 #
+# Define some functions to help get a hold of the user's Latex
+# Macros.
+#
+namePattern = ///             # The name of a macro can be either
+              ^[^a-zA-Z\d\s]$ # a single non-alphanumeric character
+              |               # or
+              ^[a-zA-Z]*$     # any number of lower and upper case
+              ///             # letters, but no numerals.
+
+valuePattern = ///^    # The stringified version of the definition
+               {.*}    # must be between brackets
+               (,\d)?  # and may be followed by a comma and a digit
+               $///
+
+getUserMacrosPath = ->
+  userMacrosPath =  CSON.resolve(path.join(atom.getConfigDirPath(),'markdown-preview-plus'))
+  userMacrosPath ? path.join(atom.getConfigDirPath(),'markdown-preview-plus.cson')
+
+loadMacrosFile = (filePath) ->
+  return {} unless CSON.isObjectPath(filePath)
+  CSON.readFileSync filePath, (error, object={}) =>
+    if error?
+      console.warn "Error reading Latex Macros file '#{filePath}': #{error.stack ? error}"
+      atom.notifications?.addError("Failed to load Latex Macros from '#{filePath}'", {detail: error.message, dismissable: true})
+    object
+
+loadUserMacros = ->
+  userMacrosPath = getUserMacrosPath()
+  if fs.isFileSync(userMacrosPath)
+    result = loadMacrosFile(userMacrosPath)
+  else
+    console.log "Creating markdown-preview-plus.cson, this is a one-time operation."
+    createMacrosTemplate(userMacrosPath)
+    result = loadMacrosFile(userMacrosPath)
+
+createMacrosTemplate = (filePath) ->
+  templatePath = path.join(__dirname,"../assets/macros-template.cson")
+  templateFile = fs.readFileSync templatePath, 'utf8'
+  fs.writeFileSync filePath, templateFile
+
+checkMacros = (macrosObject) ->
+  for name, value of macrosObject
+    unless name.match(namePattern) and "#{value}".match(valuePattern)
+      delete macrosObject[name]
+      atom.notifications?.addError("Failed to load LaTeX macro named '#{name}'. Please see the [LaTeX guide](https://github.com/Galadirith/markdown-preview-plus/blob/master/LATEX.md#macro-names)", {dismissable: true})
```

Comment:
This links to my edits on `LaTeX.md`, I'm pretty sure I got the link right, but it's hard to preview until this is published.